#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_format-errors.dpatch by Brian Thomason <brian.thomason@canonical.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix some string formatting issues

--- swftools-0.9.0.bak/lib/pdf/GFXOutputDev.cc	2009-04-12 19:25:32.000000000 -0400
+++ swftools-0.9.0/lib/pdf/GFXOutputDev.cc	2010-04-06 17:36:05.488911064 -0400
@@ -2504,8 +2504,8 @@
     memset(f, 0, sizeof(fontfile_t));
     f->filename = filename;
     int len = strlen(filename);
-    char*r1 = strrchr(filename, '/');
-    char*r2 = strrchr(filename, '\\');
+    const char*r1 = strrchr(filename, '/');
+    const char*r2 = strrchr(filename, '\\');
     if(r2>r1)
         r1 = r2;
     if(r1) {
--- swftools-0.9.0.bak/lib/pdf/pdf.cc	2009-04-12 19:25:32.000000000 -0400
+++ swftools-0.9.0/lib/pdf/pdf.cc	2010-04-06 17:46:40.516910957 -0400
@@ -458,7 +458,7 @@
     i->filename = strdup(filename);
 
     char*x = 0;
-    if((x = strchr(filename, '|'))) {
+    if((x = (char *) strchr(filename, '|'))) {
 	*x = 0;
 	userPassword = x+1;
     }
